packages/builder/src/components/automation/SetupPanel/SchemaSetup.svelte

```diff
@@ -0,0 +1,113 @@
+<script>
+  import { Input } from "@budibase/bbui"
+
+  export let value = {}
+  $: fieldsArray = Object.entries(value).map(([name, type]) => ({
```

Comment:
NAB: I think you might save yourself a lot of pain here if you just dealt with the array the whole time, used `bind`, and construct/reduce the schema object from the array when the actual action gets executed.